src/CacheInterface.php

```diff
@@ -4,31 +4,40 @@
 
 use React\Promise\PromiseInterface;
 
+/**
+ * All methods only reject with an exception when an error occurs on the underlying storage
+ * layer, for example the connection with Redis dropped and cannot be recovered, or the
+ * directory on the filesystem used to store cache has the wrong permissions or is full.
+ */
 interface CacheInterface
 {
     /**
      * Retrieve an item from the cache, resolves with its value on
-     * success or null when no item can be found.
+     * success or null when no item can be found. It will reject with an exception
+     * on error. (Note that a cache miss isn't an error.)
      *
      * @param string $key
      * @return PromiseInterface
      */
     public function get($key);
 
     /**
-     * Store an item in the cache.
+     * Store an item in the cache, returns a promise which resolves to true on success or
+     * false on error.
      *
      * @param string $key
      * @param mixed $value
-     * @return void
+     * @return PromiseInterface<bool>
```

Comment:
I don't see any notation for this in any of the major projects, so I would suggest using the more verbose form `@return PromiseInterface Returns a promise which resolves to true on success of false on error` and possibly update the description above? (See also below)